modules/test/playwright/pages/calendar-web/ModalRecurrencePage.ts

```diff
@@ -0,0 +1,53 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2000 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Locator, Page} from '@playwright/test';
+
+export class ModalRecurrencePage {
+	readonly afterRadio: Locator;
+	readonly countTextbox: Locator;
+	readonly doneButton: Locator;
+	page: Page;
+	readonly repeatSelect: Locator;
+	readonly wednesdayCheckbox: Locator;
+
+	constructor(page: Page) {
+		this.afterRadio = page
+			.frameLocator('iframe')
+			.locator('input[type="radio"][value="after"]');
+		this.countTextbox = page
+			.frameLocator('iframe')
+			.getByRole('textbox', {name: 'Count'});
+		this.doneButton = page
+			.frameLocator('iframe')
+			.getByRole('button', {name: 'Done'});
+		this.page = page;
+		this.repeatSelect = page
+			.frameLocator('iframe')
+			.locator('select[title="frequency"]');
+		this.wednesdayCheckbox = page
+			.frameLocator('iframe')
+			.getByRole('checkbox', {
+				exact: true,
+				name: 'Wednesday',
+			});
+	}
+
+	async addRecurrence(recurrence: Recurrence) {
+		const {frequency, ocurrences, repeatDays} = recurrence;
+
+		await this.repeatSelect.selectOption(frequency);
+
+		if (repeatDays.includes('Wednesday')) {
```

Comment:
`addRecurrence` should be generic, why are we dealing with specific days inside this method?

Declare specific methods to deal with the selection of days:

```
	async handleDayCheckboxesSelection(repeatDays: daysOfTheWeek[], checked: boolean) {
		for (const day of repeatDays) {
			await this.page.frameLocator('iframe')
				.getByRole('checkbox', {
					exact: true,
					name: day,
				}).setChecked(checked);;
		}
	}

	async selectRepeatDays(repeatDays: daysOfTheWeek[]) {
		await this.handleDayCheckboxesSelection(repeatDays, true);
	}

	async unselectRepeatDays(repeatDays: daysOfTheWeek[]) {
		await this.handleDayCheckboxesSelection(repeatDays, false);
	} 
``` 

Replace this if statement with await this.selectRepeatDays(repeatDays)`